server/src/main/java/io/crate/role/Privilege.java

```diff
@@ -58,12 +58,12 @@ public static Privilege fromXContent(XContentParser parser) throws IOException {
                 String currentFieldName = parser.currentName();
                 currentToken = parser.nextToken();
                 switch (currentFieldName) {
-                    case "state":
+                    case "policy", "state":
                         if (currentToken != XContentParser.Token.VALUE_NUMBER) {
                             throw new ElasticsearchParseException(
                                 "failed to parse privilege, 'state' value is not a number [{}]", currentToken);
                         }
-                        state = PrivilegeState.values()[parser.intValue()];
+                        policy = Policy.values()[parser.intValue()];
```

Comment:
minor: not related to your change, but afaik, a `Enum.values()` call will result in a copy of the backing array. So better store `Policy.values()` outside of the loop? This was also the reason to have a static `VALUES` cache inside the enum, but I also think this my micro optimization for small enums. So I'm fine to remove it, but we should avoid calling it inside a loop, I think.

Reply:
Great, didn't know and read about it. Thx a lot!
I'm reintroducing it, so we use the cache for `Privilege.fromXContent()`.